### PR TITLE
updateOne & updateMany fixes

### LINKED ISSUE #26: Fix updateOne and updateMany field mappings
Fix updateOne and updateMany field mappings based on the server implementation

https://github.com/stargate/jsonapi/issues/168
https://github.com/stargate/stargate-mongoose/discussions/24

### PR DESCRIPTION
**Pull request by kathirsvn**
**What this PR does**:
Handled a scenario where server will return `moreData` flag as `true` in `updateMany` API response
Tested and raised issues with JSON API for `upsert` flag in `updateOne` and `updateMany` APIs

**Which issue(s) this PR fixes**:
Fixes #26 

**Checklist**
- [X] Changes manually tested
- [X] Automated Tests added/updated
- [X] Documentation added/updated
- [X] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/collections/collection.ts [hunk 1]
```diff
@@ -253,5 +256,12 @@ export class Collection {
   async dropIndexes(index: any, options: any) {
     throw new Error('Not Implemented');
   }
+}
 
+export class StargateMongooseError extends Error {
+  message: string;
+  constructor(message: any) {    
+    super(`Operation failed with the following error: ${message}`);
```
Comment (vkarpov15) [inline src/collections/collection.ts]: Let's provide more info on the operation that failed. `Operation ${operationName} failed with the following error:`.

Also, add a `command` property to `StargateMongooseError` that contains the command that failed.

File: tests/collections/collection.test.ts [hunk 2]
```diff
@@ -383,6 +429,95 @@ for (const testClient in testClients) {
...
+        
+        //const idToUpdateAndCheck = sampleDocsWithIdList[0]._id;
+        try{
+          const updateManyResp = await collection.updateMany({"city": "nyc"},
+                                                    {
+                                                      "$set" : { "state" : "ny" }
+                                                    });
+        }
+        catch(e: any){
+          assert.ok(e);
```
Comment (vkarpov15) [inline tests/collections/collection.test.ts]: Unfortunately, this test will pass if no error is thrown. There are a couple of patterns we recommend for asserting failures with async/await with JavaScript:

1) Golang-style error handling.

```
const err = await collection.updateMany(filter, update).then(() => null, err => err);
assert.ok(err);
assert.strictEqual(err.message, "...");
```

2) `assert.rejects()`

```javascript
await assert.rejects(collection.updateMany(filter, update), /More than 20 records found/);
```

3) Throw at the end of the `try` block

```javascript
try {
  await collection.updateMany(filter, update);
  assert.ok(false);
} catch (err) {
  // ...
}
```

(3) requires the least amount of changes from the current state of the code, but I find (1) to be the easiest one to work with. What do you think @kathirsvn ?

File: tests/collections/collection.test.ts [hunk 3]
```diff
@@ -383,6 +429,95 @@ for (const testClient in testClients) {
...
+          doc.username = doc.username+(index+1);
+        });
+        const res = await collection.insertMany(docList);    
+        assert.strictEqual(res.insertedCount, docList.length);
+        assert.strictEqual(res.acknowledged, true);
+        assert.strictEqual(_.keys(res.insertedIds).length, 25);
+        
+        //const idToUpdateAndCheck = sampleDocsWithIdList[0]._id;
+        try{
+          const updateManyResp = await collection.updateMany({"city": "nyc"},
```
Comment (vkarpov15) [inline tests/collections/collection.test.ts]: Question: are the updates "rolled back" if `moreData` is returned? Or does stargate update 20 documents, and then return `moreData` to tell the client that it couldn't update everything in one shot?

Comment (kathirsvn) [reply tests/collections/collection.test.ts]: It's the later. Stargate updates 20 documents and then returns `moreData` as `true` if it finds more records to be updated.

File: tests/collections/collection.test.ts (hunk 2 above)
Comment (kathirsvn) [reply tests/collections/collection.test.ts]: Good catch @vkarpov15 I cleaned up, all the exception based tests in [this PR](https://github.com/stargate/stargate-mongoose/pull/55/files) I missed this and ll fix this as well. Basically, I used the 3rd option you mentioned above on that PR.

File: tests/collections/collection.test.ts (hunk 3 above)
Comment (vkarpov15) [reply tests/collections/collection.test.ts]: Thanks for clarifying.

File: src/collections/collection.ts (hunk 1 above)
Comment (kathirsvn) [reply src/collections/collection.ts]: Fixed this.

**Review by vkarpov15 — APPROVED** [review]

**Pull request merged**